dot/core/state.go

```diff
@@ -33,6 +33,7 @@ type BlockState interface {
 	BestBlock() (*types.Block, error)
 	SubChain(start, end common.Hash) ([]common.Hash, error)
 	AddBlock(*types.Block) error
+	GetAllHashesForParentDepth(*types.Header) (map[common.Hash]*big.Int, error)
```

Comment:
name is unclear, we are getting nodes at the same depth as the provided header.

should also return just an array of hashes, not a map

Reply:
updated